test/spec/struct.wast

```diff
@@ -105,3 +105,29 @@
   )
   "invalid rtt"
 )
+
+(assert_invalid
+  (module
+    (type $vec (struct (field (ref func))))
+    (func $test
+      (drop
+        ;; too many arguments
+        (struct.new_with_rtt $vec (i32.const 1) (i32.const 2) (rtt.canon $vec))
+      )
+    )
+  )
+  "invalid numbe of arguments to struct.new"
+)
+
+(assert_invalid
+  (module
+    (type $vec (struct (field (ref func)) (field f64)))
+    (func $test
+      (drop
+        ;; too few arguments
+        (struct.new_with_rtt $vec (i32.const 1) (rtt.canon $vec))
```

Comment:
The number of fields seems to be the same? But the type looks different.

Reply:
The RTT field is not counted. So there is just one argument to the struct here.

Reply:
Yeah I didn't count that either. But the `$vec` struct has one field and this has one field too. The test above has two fields and it has too many. And this has one field and this is too few?

Reply:
Oh sorry, this time `$vec` is different... I only looked at `$vec` in the first test. Sorry nevermind.